facebook/OpenBIC
# Common: Support different sensor polling time

Summary:
- Support different sensor can set different polling times.
- Add polling time to sensor config table of Yv35 CL, Yv35 BB, Yv35 RF, and GT CC.
- Change P3V_BAT polling time of CL BIC from 1 second to 1 hour.

Test pan:
- Build code: Pass
- Check P3V_BAT poll time: Pass
- Get sensor reading: Pass

Log
1. Check P3V_BAT poll time is 1 hour via debug print.
uart:~$ get_sensor_reading 0x21 access_time 1
[00:00:00.525,000] <inf> usb_cdc_acm: Device resumed
[00:00:00.525,000] <inf> usb_cdc_acm: from suspend
[00:00:00.861,000] <inf> usb_cdc_acm: Device configured
[00:00:00.525,000] <inf> usb_cdc_acm: Device resumed
[00:00:00.525,000] <inf> usb_cdc_acm: from suspend
[00:00:00.861,000] <inf> usb_cdc_acm: Device configured
uart:~$ BIC Ready

uart:~$ get_sensor_reading 0x21 access_time 3602
get_sensor_reading 0x21 access_time 7202
get_sensor_reading 0x21 access_time 10802
get_sensor_reading 0x21 access_time 14403
get_sensor_reading 0x21 access_time 18003
get_sensor_reading 0x21 access_time 21604

2. Get reading of all sensors.
- CL BIC
root@bmc-oob:~# sensor-util slot3
slot3:
MB Inlet Temp                (0x1) :   30.00 C     | (ok)
MB Outlet Temp               (0x2) :   42.00 C     | (ok)
FIO Temp                     (0x3) :   30.00 C     | (ok)
PCH Temp                     (0x4) :   42.00 C     | (ok)
SOC CPU Temp                 (0x5) :   37.00 C     | (ok)
SOC Therm Margin             (0x14) :  -63.00 C     | (ok)
CPU TjMax                    (0x15) :  100.00 C     | (ok)
DIMMA Temp                   (0x6) :   38.00 C     | (ok)
DIMMC Temp                   (0x7) :   37.00 C     | (ok)
DIMMD Temp                   (0x9) :   36.00 C     | (ok)
DIMME Temp                   (0xA) :   37.00 C     | (ok)
DIMMG Temp                   (0xB) :   35.00 C     | (ok)
DIMMH Temp                   (0xC) :   33.00 C     | (ok)
SSD0 Temp                    (0xD) :   33.00 C     | (ok)
HSC Temp                     (0xE) :   38.33 C     | (ok)
VCCIN SPS Temp               (0xF) :   50.00 C     | (ok)
FIVRA SPS Temp               (0x10) :   46.00 C     | (ok)
EHV SPS Temp                 (0x11) :   39.00 C     | (ok)
VCCD SPS Temp                (0x12) :   40.00 C     | (ok)
FAON SPS Temp                (0x13) :   42.00 C     | (ok)
P12V_STBY Vol                (0x20) :   12.75 Volts | (ok)
P3V_BAT Vol                  (0x21) :    3.13 Volts | (ok)
P3V3_STBY Vol                (0x22) :    3.34 Volts | (ok)
P1V8_STBY Vol                (0x23) :    1.79 Volts | (ok)
P1V05_PCH Vol                (0x24) :    1.06 Volts | (ok)
P5V_STBY Vol                 (0x25) :    4.94 Volts | (ok)
P12V_DIMM Vol                (0x26) :   12.53 Volts | (ok)
P1V2_STBY Vol                (0x27) :    1.19 Volts | (ok)
P3V3_M2 Vol                  (0x28) :    3.29 Volts | (ok)
HSC Input Vol                (0x29) :   12.49 Volts | (ok)
VCCIN VR Vol                 (0x2A) :    1.77 Volts | (ok)
FIVRA VR Vol                 (0x2C) :    1.82 Volts | (ok)
EHV VR Vol                   (0x2D) :    1.80 Volts | (ok)
VCCD VR Vol                  (0x2E) :    1.14 Volts | (ok)
FAON VR Vol                  (0x2F) :    1.05 Volts | (ok)
HSC Output Cur               (0x30) :    8.40 Amps  | (ok)
VCCIN VR Cur                 (0x31) :   33.70 Amps  | (ok)
FIVRA VR Cur                 (0x32) :    3.90 Amps  | (ok)
EHV VR Cur                   (0x33) :    0.80 Amps  | (ok)
VCCD VR Cur                  (0x34) :    2.70 Amps  | (ok)
FAON VR Cur                  (0x35) :    8.80 Amps  | (ok)
CPU PWR                      (0x38) :   78.00 Watts | (ok)
HSC Input Pwr                (0x39) :  102.88 Watts | (ok)
VCCIN VR Pout                (0x3A) :   59.00 Watts | (ok)
FIVRA VR Pout                (0x3C) :    7.00 Watts | (ok)
EHV VR Pout                  (0x3D) :    1.00 Watts | (ok)
VCCD VR Pout                 (0x3E) :    3.00 Watts | (ok)
FAON VR Pout                 (0x3F) :    9.00 Watts | (ok)
DIMMA PMIC_Pout              (0x1E) :    0.12 Watts | (ok)
DIMMC PMIC_Pout              (0x1F) :    0.00 Watts | (ok)
DIMMD PMIC_Pout              (0x36) :    0.00 Watts | (ok)
DIMME PMIC_Pout              (0x37) :    0.00 Watts | (ok)
DIMMG PMIC_Pout              (0x42) :    0.12 Watts | (ok)
DIMMH PMIC_Pout              (0x47) :    0.00 Watts | (ok)

- RF BIC
root@bmc-oob:~# sensor-util slot3 | grep RF
RF MB Inlet Temp             (0x50) :   28.00 C     | (ok)
RF CXL_cntr_Temp             (0x52) :   29.00 C     | (ok)
RF P0V9_A Temp               (0x58) :   36.00 Volts | (ok)
RF P0V8_A Temp               (0x59) :   34.00 Volts | (ok)
RF P0V8_D Temp               (0x5A) :   32.00 Volts | (ok)
RF PVDDQ_AB Temp             (0x5B) :   35.00 Volts | (ok)
RF PVDDQ_CD Temp             (0x5C) :   36.00 Volts | (ok)
RF P12V_STBY Vol             (0x5D) :   12.48 Volts | (ok)
RF P3V3_STBY Vol             (0x5E) :    3.33 Volts | (ok)
RF P5V_STBY Vol              (0x5F) :    5.12 Volts | (ok)
RF P1V2_STBY Vol             (0x60) :    1.19 Volts | (ok)
RF P1V8_ASIC Vol             (0x61) :    1.80 Volts | (ok)
RF P0V9_A Vol                (0x62) :    1.08 Volts | (unr)
RF P0V8_A Vol                (0x64) :    0.82 Volts | (ok)
RF P0V8_D Vol                (0x65) :    0.82 Volts | (ok)
RF PVDDQ_AB Vol              (0x66) :    1.20 Volts | (ok)
RF PVDDQ_CD Vol              (0x67) :    1.20 Volts | (ok)
RF PVPP_AB Vol               (0x68) :    2.50 Volts | (ok)
RF PVPP_CD Vol               (0x69) :    2.50 Volts | (ok)
RF PVTT_AB Vol               (0x6A) :    0.60 Volts | (ok)
RF PVTT_CD Vol               (0x6B) :    0.60 Volts | (ok)
RF P12V_STBY Cur             (0x6C) :    0.25 Volts | (ok)
RF P3V3_STBY Cur             (0x6D) :    0.16 Volts | (ok)
RF P0V9_A Cur                (0x6E) :    0.20 Volts | (ok)
RF P0V8_A Cur                (0x6F) :    0.20 Volts | (ok)
RF P0V8_D Cur                (0x70) :    0.30 Volts | (ok)
RF PVDDQ_AB Cur              (0x71) :   -0.20 Volts | (ok)
RF PVDDQ_CD Cur              (0x72) :   -0.40 Volts | (ok)
RF P12V_STBY_PWR             (0x73) :    3.12 Volts | (ok)
RF P3V3_STBY_PWR             (0x74) :    0.52 Volts | (ok)
RF P0V9_A_PWR                (0x75) :   -1.00 Volts | (ok)
RF P0V8_A_PWR                (0x76) :   -1.00 Volts | (ok)
RF P0V8_D_PWR                (0x77) :    0.00 Volts | (ok)
RF PVDDQ_AB_PWR              (0x78) :   -1.00 Volts | (ok)
RF PVDDQ_CD_PWR              (0x79) :    0.00 Volts | (ok)

- BB BIC
root@bmc-oob:~# sensor-util slot1 | grep BB
BB_INLET_TEMP                (0xD1) :   27.00 C     | (ok)
BB_OUTLET_TEMP               (0xD2) :   30.00 C     | (ok)
BB_HSC_TEMP                  (0xD3) :   26.43 C     | (ok)
BB_P5V_STBY                  (0xD4) :    5.14 Volts | (ok)
BB_P12V_STBY                 (0xD5) :   12.56 Volts | (ok)
BB_P3V3_STBY                 (0xD6) :    3.37 Volts | (ok)
BB_P5V_USB                   (0xE9) :    5.14 Volts | (ok)
BB_P1V2_STBY                 (0xD9) :    1.21 Volts | (ok)
BB_P1V0_STBY                 (0xDA) :    1.01 Volts | (ok)
BB_MEDUSA_VIN                (0xDB) :   12.40 Volts | (ok)
BB_MEDUSA_VOUT               (0xD7) :   12.40 Volts | (ok)
BB_HSC_VIN                   (0xDC) :   12.38 Volts | (ok)
BB_MEDUSA_CUR                (0xDF) :    5.08 Amps  | (ok)
BB_HSC_IOUT                  (0xDE) :    6.07 Amps  | (ok)
BB_FAN_IOUT                  (0xE8) :    2.75 Amps  | (ok)
BB_MEDUSA_PWR                (0xD8) :   62.97 Watts | (ok)
BB_HSC_PIN                   (0xDD) :   75.00 Watts | (ok)
BB_HSC_EIN                   (0xEA) :   72.98 Watts | (ok)
BB_HSC_PEAK_IOUT             (0xEB) :   16.58 Amps  | (ok)
BB_HSC_PEAK_PIN              (0xEC) :  201.86 Watts | (ok)
